gotfparse/pkg/converter/converter.go

```diff
@@ -36,118 +30,214 @@ type terraformMeta struct {
 func (t *terraformConverter) VisitJSON() *gabs.Container {
 	jsonOut := gabs.New()
 
-	var visitor func(b *terraform.Block, parentKey string)
+	for _, m := range t.modules {
+		t.visitModule(m, jsonOut)
+	}
 
-	visitor = func(b *terraform.Block, parentKey string) {
-		blockName := b.GetMetadata().String()
+	return jsonOut
+}
 
-		blockJSON := gabs.New()
-		arrayKey := blockName
+func (t *terraformConverter) shouldWalk(b *terraform.Block) bool {
```

Comment:
I won't comment on each one, but please add doc strings for all of the new functions and methods.

As an example question I'd find my future self asking:

Why does should walk skip locals, output, terraform, variable?
What is a raw value? How is it different from a value?
What is a forEachCount?

- shouldWalk
- buildBlock
- visitBlock
- toRawValue
- fromValuetoRawValue
- getForEachCount
- visitModule
- handleDynamicBlocks